File: rebuild/validation/repository-intelligence/polyglot-structural/fixtures/cpp/src/greeter.cpp
#include "greeter.hpp"

#include <utility>

namespace sample {

Greeter::Greeter(std::string prefix) : prefix_(std::move(prefix)) {}

std::string Greeter::name() const {
    return prefix_;
}

std::string Greeter::greet(const Identifier &person) const {
    return name() + ", " + person;
}

}  // namespace sample

File: rebuild/validation/repository-intelligence/polyglot-structural/fixtures/java/src/main/java/example/Greeter.java
package example;

import java.util.Objects;

interface Named {
    String name();
}

class Greeter implements Named {
    private final String prefix;

    Greeter(String prefix) {
        this.prefix = Objects.requireNonNull(prefix);
    }

    public String name() {
        return prefix;
    }

    String greet(String person) {
        return name() + ", " + person;
    }
}
